apps/etterlatte-trygdetid-kafka/src/main/kotlin/MigreringHendelser.kt

```diff
@@ -41,51 +41,48 @@ internal class MigreringHendelser(rapidsConnection: RapidsConnection, private va
-            trygdetidService.beregnTrygdetid(behandlingId).also {
-                logger.info("Oppretta trygdetid for behandling $behandlingId")
-            }
-        }.takeIf { it.isSuccess }
-            ?.let {
-                withFeilhaandtering(packet, context, Migreringshendelser.TRYGDETID_GRUNNLAG) {
-                    val behandlingId = packet.behandlingId
-                    val request = packet.hendelseData
-                    logger.info("Oppretter grunnlag for trygdetid for $behandlingId")
+        val behandlingId = packet.behandlingId
+        val request = packet.hendelseData
 
-                    try {
-                        // Oppretter alle trygdetidsperioder og beholder siste beregnede trygdetid
-                        val beregnetTrygdetid =
-                            request.trygdetid.perioder.map { periode ->
-                                trygdetidService.beregnTrygdetidGrunnlag(behandlingId, tilGrunnlag(periode))
-                            }.lastOrNull() ?: it.getOrThrow()
+        logger.info("Mottatt trygdetid-migreringshendelse for behandling $behandlingId")
+        trygdetidService.beregnTrygdetid(behandlingId)
+        logger.info("Opprettet trygdetid for behandling $behandlingId, forsøker å legge til perioder")
 
-                        sendBeregnetTrygdetid(packet, beregnetTrygdetid, context, behandlingId)
-                    } catch (e: Exception) {
-                        logger.warn(
-                            "Problem ved opprettelse av trygdetidsperioder for behandling $behandlingId - overstyrer beregnet trygdetid",
-                            e,
+        val oppdatertTrygdetid: TrygdetidDto =
+            if (request.trygdetid.perioder.isNotEmpty()) {
+                try {
+                    request.trygdetid.perioder.map { periode ->
+                        val grunnlag = tilGrunnlag(periode)
+                        logger.info(
+                            "Forsøker å legge til periode ${grunnlag.periodeFra}-${grunnlag.periodeTil} for behandling $behandlingId",
                         )
-
-                        // TODO - EY-2602 - flere felter trengs - feks utland
-                        val detaljertBeregnetTrygdetid =
-                            DetaljertBeregnetTrygdetidResultat.fraSamletTrygdetidNorge(
-                                request.beregning.anvendtTrygdetid.toInt(),
-                            )
-
-                        val beregnetTrygdetid =
-                            trygdetidService.overstyrBeregnetTrygdetid(
-                                behandlingId,
-                                detaljertBeregnetTrygdetid,
-                            )
-
-                        sendBeregnetTrygdetid(packet, beregnetTrygdetid, context, behandlingId)
-                    }
+                        trygdetidService.beregnTrygdetidGrunnlag(behandlingId, grunnlag)
+                    }.last()
```

Comment:
Vil trygdetid alltid være sortert slik at man faktisk får siste trygdetid her?

Reply:
Den tjenesten tar inn en trygdetidsperiode og reberegner hele den totale trygdetiden (samlet faktisk, fremtidig og sum av de to), så rekkefølgen spiller ikke noe rolle her. Det vi ønsker er resultatet fra det siste kallet.